api/v1alpha1/collectors.go

```diff
@@ -0,0 +1,27 @@
+package v1alpha1
+
+// Collector represents a reference to a Collector to be executed as part of the release workflow.
+// +kubebuilder:object:generate=true
+type Collector struct {
+	// Name of the collector
+	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
+	// +required
+	Name string `json:"name"`
+
+	// Collector is the type of collector to be used
+	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
+	// +required
+	Collector string `json:"collector"`
+
+	// Params is a slice of parameters for a given collector
+	Params []Param `json:"params"`
+}
+
+// Param represents a parameter for a collector
+type Param struct {
```

Comment:
Can't we use the same Param definition we have in the tekton directory? Instead of defining an identical struct in two places

Reply:
I didn't want to link this with the Tekton stuff...

Probably we could move it to operator-toolkit in the future.

Reply:
I am fine with it going into operator-toolkit. I don't want two identical structs in two spots in one repo though

Reply:
Yeah, followup ticket is fine, or even adding it to the AC of the cleanup tekton dir ticket if you think that is related enough. As long as it is written on a ticket somewhere